node/server.js

```diff
@@ -5154,6 +5125,135 @@ function init_io() {
 					instance.players[NPC_prefix + npc.id] = npc;
 				}
 			} else {
+				const gMap = data.place && G.maps[data.place];
+				if (gMap && gMap.instance) {
+					// Generic instance creation and transportation
+					// data.place is the name of the instance e.g bee_dungeon
+					// data.name is the id of the instance usually a number
+
+					const instanceExists = data.name && instances[data.name] && instances[data.name].map == data.place;
+
+					if (data.name) {
+						// Player requested to enter an existing instance
+						if (!instances[data.name]) {
+							// The instance doesn't exist
+							return fail_response("transport_cant_invalid");
+						}
+					}
+
+					// Requirements for entering
+					if (gMap.enter) {
+						// Is there a place you need to be near to use the key
+						const validateRange = gMap.enter.locations && gMap.enter.locations.length > 0;
+						let inRange = !validateRange;
+
+						if (validateRange) {
+							server_log(`${gMap.enter.locations.length} locations to validate`);
+							for (const [locationsMapKey, locationType, locationIndex, range = 120] of gMap.enter.locations) {
+								if (!G.maps[locationsMapKey]) {
+									server_log(`G.maps.${locationsMapKey} does not exist`);
+									continue;
+								}
+
+								if (!G.maps[locationsMapKey][locationType]) {
+									server_log(`G.maps.${locationsMapKey}.${locationType} does not exist`);
+									continue;
+								}
+
+								if (!G.maps[locationsMapKey][locationType][locationIndex]) {
+									server_log(`G.maps.${locationsMapKey}.${locationType}.${locationIndex} does not exist`);
+									continue;
+								}
+
+								const location = G.maps[locationsMapKey][locationType][locationIndex];
+								const distanceToLocation = simple_distance(player, {
+									in: locationsMapKey,
+									map: locationsMapKey,
+									x: location[0],
+									y: location[1],
+								});
```

Comment:
This should use the `distance` function instead.

Reply:
Yes, I know but I would like to standardize it more and this is the opportunity for a little change. (Especially since you test this code on your test server which, by the way, I thank you for!)

Reply:
As far as I remember the distance function is more expensive than the `simple_distance` one, so that is why I opted to using that. I'll change it to distance though, this code should not run that often and I don't think we will safe much from using `simple_distance`

Reply:
should be solved